src/hooks/useCheckForAppUpdate.ts

```diff
@@ -0,0 +1,50 @@
+import { useSystemControllerCheckForAppUpdate } from '@baca/api/query/system/system'
+import { useTranslation } from '@baca/hooks'
+import * as Application from 'expo-application'
+import { useEffect, useState } from 'react'
+import { Platform, Alert, Linking } from 'react-native'
+
+export const useCheckForAppUpdate = () => {
+  const { t } = useTranslation()
```

Comment:
below line 7 stay only: code form line 8 and 13, rest of them: lines 9, 10 move outside the hook - above line 7,
lines 16, 17 move to const folder. Dont use setIsLoading(false), on every state of promise, read about onSettled.

Reply:
Updated, I'm really thankful for all these suggestions 😊